src/App.tsx

```diff
@@ -1,27 +1,62 @@
-import React from 'react';
-import './App.scss';
+import { useState, FC } from 'react';
 import { GoodsList } from './GoodsList';
+import { getAll, get5First, getRedGoods } from './api/goods';
+
+import { Good } from './types/Good';
+import './App.scss';
+
+enum Load {
+  None,
+  All,
+  First5,
+  Red,
+}
+
+export const App: FC = () => {
+  const [goods, setGoods] = useState([] as Good[]);
+  const [loaded, setLoaded] = useState(Load.None);
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+  const loadGoods = async (
+    goodsToLoad: () => Promise<Good[]>,
+    loadedId: Load,
+  ) => {
+    if (loadedId !== loaded) {
+      setGoods(await goodsToLoad());
+      setLoaded(loadedId);
+    }
+  };
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+  return (
+    <div className="App">
+      <h1>Dynamic list of Goods</h1>
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+      <button
+        type="button"
+        data-cy="all-button"
+        onClick={() => loadGoods(getAll, Load.All)}
+      >
+        Load all goods
+      </button>
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+      <button
+        type="button"
+        data-cy="first-five-button"
+        onClick={() => loadGoods(get5First, Load.First5)}
+      >
+        Load 5 first goods
+      </button>
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+      <button
+        type="button"
+        data-cy="red-button"
+        onClick={() => loadGoods(getRedGoods, Load.Red)}
+      >
+        Load red goods
+      </button>
 
-    <GoodsList goods={[]} />
-  </div>
-);
+      {goods.length > 0 && (
```

Comment:
this can be 

```js
{goods.length && (

Reply:
It needs to be explicit boolean cause in some cases the number shows up next to the element